documentation/milestones/individual_milestones/developer_tools.md

```diff
@@ -0,0 +1,221 @@
+  - [ ] At least 5 internal developers/maintainers have adopted and run these tools on a cloned version of the repo
+  - [ ] At least 3 open source contributors have adopted and run these tools on a forked version of the repo
+  - [ ] ADRs documenting the selection of these tools and standards have been created and approved
+  - [ ] The status of key checks are reflected as badges in a project README
+- [ ] **Automated testing framework** is live and meets the following conditions:
+  - [ ] At least 1 unit test has been added to the codebase
+  - [ ] At least 1 integration test has been added to the codebase
+  - [ ] Unit tests are run on every push to the remote GitHub repository
+  - [ ] Integration tests are run at least once before merging new code into `main`
+  - [ ] Code which fails any of the unit or integration tests will be blocked from merging into `main`
+  - [ ] A report on the percentage of code covered by tests is available after every test run
+  - [ ] Code whose test coverage falls below a certain threshold will be blocked from merging into `main`
+- [ ] **Code quality checks** are live and meet the following conditions:
+  - [ ] All checks are run on every push to the remote GitHub repository
+  - [ ] The most important checks are run on every local commit
+  - [ ] Code which fails any of these checks will be blocked from merging into `main`
+- [ ] **Security checks** are live and meet the following conditions:
+  - [ ] At least 1 (test) security vulnerability report has been submitted
+  - [ ] Maintainers are notified within 1 hour of a vulnerability being reported within the grants API codebase
+  - [ ] Maintainers are notified within 72 hours of a vulnerability being reported on an upstream dependency
+  - [ ] Security checks are running on every push to the remote GitHub repository
+  - [ ] Code which fails any of these security checks is blocked from merging into `main`
+- [ ] **Dependency management** is live and meets the following conditions:
+  - [ ] At least 2 upstream dependencies have been added to the project
+  - [ ] Compatible versions of upstream dependencies are automatically detected when a new dependency is added
+  - [ ] Maintainers are notified when a new minor or major version of an upstream dependency is available
+- [ ] **Config & secrets management** is live and meets the following conditions:
+  - [ ] At least 1 configuration variable has been added to the project
+  - [ ] This config variable has different values when it is running in different environments (e.g. dev, staging, and prod)
+  - [ ] At least 2 internal developers can access a shared value of this variable locally
+  - [ ] Developers can change the value of this config variable in their local environment without changing it for other developers
+  - [ ] An integration test has been created to validate that this variable is injected at runtime
+
+### Proposed metrics for measuring goals/value/definition of done
```

Comment:
Are there any code quality scoring tools (not just linters) we can implement to automate the calculation of metrics related to these things? 

Things like https://codeclimate.com/quality or https://www.sonarsource.com/products/sonarqube/

Reply:
That would be great to use a tool like code climate.